src/components/MDX/Sandpack/Preview.tsx

```diff
@@ -52,6 +52,12 @@ export function Preview({
     rawError = null;
   }
 
+  // When throwing a new Error in Sandpack - we want to disble the dev error dialog
+  // to show the Error Boundary fallback
+  if (rawError && rawError.message.includes('throw new Error()')) {
```

Comment:
are the demos interactive and changable? This hardcoded detection might break easily if someone throws a different error if that's the case.

Reply:
As a slightly less fragile approach, is it possible to create a custom Error class for this case?

Reply:
IMO it's ok if someone adds a different error and we show the dialog. Having this check is mostly to demonstrate the initial fallback. If they fork the code in codesandbox they'll be able to close the modal and see the fallback too. It's mostly for initial demonstration purposes. I can update this in a separate PR if we feel differently in the future.